org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/telemetry/SonarLintTelemetry.java

```diff
@@ -63,48 +63,45 @@ private static Path getOldStorageFilePath() {
   }
 
   public void optOut(boolean optOut) {
-    if (telemetryEngine != null) {
-      if (optOut == !telemetryEngine.enabled()) {
-        return;
-      }
-      telemetryEngine.enable(!optOut);
+    if (telemetry != null) {
       if (optOut) {
-        try {
-          TelemetryClientConfig clientConfig = getTelemetryClientConfig();
-          telemetryEngine.getClient().optOut(clientConfig, isAnyProjectConnected());
-        } catch (Exception e) {
-          // Silently ignore
+        if (telemetry.isEnabled()) {
+          telemetry.disable();
+        }
+      } else {
+        if (!telemetry.isEnabled()) {
+          telemetry.enable();
         }
       }
     }
   }
 
   public boolean enabled() {
-    return enabled;
-  }
-
-  public boolean optedIn() {
-    return enabled && this.telemetryEngine.enabled();
+    return telemetry != null && telemetry.isEnabled();
   }
 
   public void init() {
     if ("true".equals(System.getProperty(DISABLE_PROPERTY_KEY))) {
-      this.enabled = false;
       SonarLintLogger.get().info("Telemetry disabled by system property");
       return;
     }
     try {
-      this.telemetryEngine = new Telemetry(getStorageFilePath(), PRODUCT, SonarLintUtils.getPluginVersion());
+      TelemetryClientConfig clientConfig = getTelemetryClientConfig();
+      TelemetryClient client = new TelemetryClient(clientConfig, PRODUCT, SonarLintUtils.getPluginVersion());
+      this.telemetry = newTelemetryManager(getStorageFilePath(), client);
       SonarLintCorePlugin.getAnalysisListenerManager().addListener(this);
       this.scheduledJob = new TelemetryJob();
       scheduledJob.schedule(TimeUnit.MINUTES.toMillis(1));
-      this.enabled = true;
     } catch (Exception e) {
       // Silently ignore
-      enabled = false;
     }
   }
 
+  // visible for testing
+  public TelemetryManager newTelemetryManager(Path path, TelemetryClient client) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "path". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)